rightscript.go

```diff
@@ -835,6 +835,69 @@ func validateRightScript(file string, ignoreMissingMetadata bool) (*RightScript,
 	return &rightScript, nil
 }
 
+func rightScriptCommitLocalCopy(file string, message string) {
+	f, err := os.Open(file)
+	if err != nil {
+		fatalError("Cannot open file: %s", err.Error())
+	}
+	defer f.Close()
+
+	metadata, err := ParseRightScriptMetadata(f)
+	if err != nil {
+		fatalError("Cannot parse RightScript: %s metadata: %s", file, err.Error())
+	}
+	var rightScriptName string
+
+	if metadata == nil {
+		scriptName := path.Base(file)
+		scriptExt := path.Ext(scriptName)
+		rightScriptName = strings.TrimRight(scriptName, scriptExt)
+	} else {
+		rightScriptName = metadata.Name
+	}
+	if rightScriptName != "" {
+		rightScriptCommit(rightScriptName, message)
+	} else {
+		fatalError("Could not retrieve RightScript name in %s", file)
+	}
+}
+
+func rightScriptCommitRemoteCopy(href, message string) {
+	client, _ := Config.Account.Client15()
+
+	rightscriptLocator := client.RightScriptLocator(href)
+
+	rightscript, err := rightscriptLocator.Show(rsapi.APIParams{"view": "inputs_2_0"})
+	if err != nil {
+		fatalError("Could not find RightScript with href %s: %s", href, err.Error())
+	}
+	rightScriptCommit(rightscript.Name, message)
+}
+
+func rightScriptCommit(rightScriptName, message string) {
+	foundID, err := rightScriptIdByName(rightScriptName)
```

Comment:
I think the way that we should split these up is a function that makes a commit based on an HREF and another function that finds the RightScript by name and then calls the first function. That way we don't need to make extra API calls if we are committing by ID/HREF.